### ISSUE TITLE
Flicker free loading

### ISSUE THREAD

**Issue by mythmon** [mozilla/corsica#36]
The flicker of new content loading is annoying and creates distraction.

Things that don't work:
- Doing nothing
- Waiting until the iframe emits a "load" event before showing it. The iframe emits this event when it i loaded, not when it's contents finish loading (such as images).

**Comment by lonnen** [comment]
other idea:

* inject a content script that listens for the `load` event, then fires a `postMessage` to the parent frame to display it

**Comment by mythmon** [comment]
I don't think we have anyway to inject scripts, unless we start modifying the browser. That being said, it would be pretty easy to have an addon that did that. The difficulty would be making Corsica work both with and without the addon.

**Comment by lonnen** [comment]
That makes total sense and I've been thinking in terms of add-ons too long. Cross domains rules prevent this from happening in any useful way.
